### PR TITLE
fix: mistake in updating invoice

### PR DESCRIPTION
**Pull request by JustSamuel**
# Description
Turns out I do monkey paste.

## Types of changes

- Bug fix _(non-breaking change which fixes an issue)_

### CODE REVIEW

File: src/service/invoice-service.ts [hunk 1]
```diff
@@ -397,16 +397,26 @@ export default class InvoiceService {
...
+          postalCode,
+          city,
+          country,
+          reference,
+          attention,
+          date: date ? new Date(date) : undefined,
+          // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        }).filter(([_, value]) => value !== undefined),
+      ),
+    );
```
Comment (Yoronex) [inline src/service/invoice-service.ts:417]: Why not just do `await Invoice.update({ id: base.id }, update)`? You only still need to parse the date to an actual date, but that method looks much cleaner and easier to understand to me.

**Review by Yoronex — CHANGES_REQUESTED** [review]

File: src/service/invoice-service.ts (hunk 1 above)
Comment (JustSamuel) [reply src/service/invoice-service.ts:417]: You would still need to filter undefinea

Comment (JustSamuel) [reply src/service/invoice-service.ts:417]: Heb een nieuwe suggestie gedaan voor je

Comment (Yoronex) [reply src/service/invoice-service.ts]: But is `UpdateInvoiceParams` not a "subset" of `Invoice`? Then you can just use `Invoice.update` method, as undefined objects should be ignored? Only if they are `null`, they should be set to `null` in the database. For any attributes that still need to be parsed (like date), you can use the spread operator, like

```
const { date, ...props } = update;
await Invoice.update({ id: base.id }, { ...props, date: date ? new Date(date) : undefined });
```

Comment (JustSamuel) [reply src/service/invoice-service.ts]: that works, thanks!

**Review by Yoronex — APPROVED** [review]

**Pull request merged**